src/p2p_plugin.c

```diff
@@ -81,20 +85,49 @@ ncclResult_t pluginInit(ncclDebugLogger_t logFunction)
   }
   switch (p2p_plugin) {
     case NCCL_P2P_IB:
-      NCCL_PLUGIN_SYMBOL = ibPlugin;
+      ncclNetPlugin_v6 = ibPlugin_v6;
+      ncclNetPlugin_v5 = ibPlugin_v5;
+      ncclNetPlugin_v4 = ibPlugin_v4;
       break;
 #ifdef HAVE_UCX_PLUGIN
     case NCCL_P2P_UCX:
-      NCCL_PLUGIN_SYMBOL = ucxPlugin;
+      ncclNetPlugin_v6 = ucxPlugin_v6;
+      ncclNetPlugin_v5 = ucxPlugin_v5;
+      ncclNetPlugin_v4 = ucxPlugin_v4;
       break;
     case NCCL_P2P_UCX_RMA:
-      NCCL_PLUGIN_SYMBOL = ucxRmaPlugin;
+      ncclNetPlugin_v6 = ucxRmaPlugin_v6;
+      ncclNetPlugin_v5 = ucxRmaPlugin_v5;
+      ncclNetPlugin_v4 = ucxRmaPlugin_v4;
       break;
 #endif
   }
-  INFO(NCCL_INIT|NCCL_NET, "P2P plugin %s", NCCL_PLUGIN_SYMBOL.name);
 
-  return NCCL_PLUGIN_SYMBOL.init(logFunction);
+  if (version == 6) {
+    INFO(NCCL_INIT|NCCL_NET, "P2P plugin %s", ncclNetPlugin_v6.name);
+    return ncclNetPlugin_v6.init(logFunction);
+  } else if (version == 5) {
+    INFO(NCCL_INIT|NCCL_NET, "P2P plugin %s", ncclNetPlugin_v5.name);
+    return ncclNetPlugin_v5.init(logFunction);
+  } else if (version == 4) {
+    INFO(NCCL_INIT|NCCL_NET, "P2P plugin %s", ncclNetPlugin_v4.name);
+    return ncclNetPlugin_v4.init(logFunction);
+  } 
+}
```

Comment:
Not sure you need that switch... aren't all plugin names always going to be the same? In which case you could just use the latest version's name and avoid passing the version to pluginInit().